docs/policy/memtier.md

```diff
@@ -1,70 +1,314 @@
+and the memory zones in the die, sockets end up containing the CPU cores and
+memory zones in the socket's dies, and the root ends up containing all CPU
+cores and memory zones in all sockets. The actual pool tree is built with all
+non-branching paths eliminated, by omitting any child node which would have
+ended up being the lone single child of its parent. This is merely an minor
+optimisation in the implementation and it has no practical significance for
+this discussion.
+
+With this setup, each pool in the tree has a topologically aligned set of CPU
+and memory resources. The amount of available resources gradually increases in
+the tree from bottom to top, while the strictness of alignment is gradually
+relaxed. In other words, as one moves from bottom to top in the tree, it is
+getting gradually easier to fit in a workload, but the price paid for this is
+a gradually increasing maximum potential cost or penalty for memory access and
+data transfer between CPU cores.
+
+Another property of this setup is that the resources sets of sibling pools at
+the same depth in the tree are disjoint while the resource sets of descendant
+pools along the same path in the tree partially overlap, with the intersection
+decreasing as the the distance between pools increases. This makes it easy to 
+isolate workloads from each other. As long as workloads are assigned to pools
+which has no other common ancestor than the root, the resources of these
+workloads should be as well isolated from each other as possible on the given
+hardware.
+
+With such an arrangement, the memtier policy should handle topology-aware
+alignment of resources without any special or extra configuration. When
+allocating resources, the policy
+
+  - filters out all pools with insufficient free capacity
+  - runs a scoring algorithm for the remaining ones
+  - picks the one with the best score
+  - assigns resources to the workload from there
+
+Although the details of the scoring algorithm are subject to change as the
+implementation evolves, its basic principles are roughly
+
+  - prefer pools lower in the tree, IOW stricter alignment and lower latency
+  - prefer idle pools over busy ones, IOW more remaining free capacity and
+    fewer workloads
+  - prefer pools with better overall device alignment
+
+## Features
+
+The memtier policy has the following features:
+
+  - topologically aligned allocation of CPU and memory
+    * assign CPU and memory to a workloads with tightest available alignment
+  - aligned allocation of devices
+    * pick pool for workload based on localoty of devices already assigned
+  - shared allocation of CPU cores
+    * assign workload to shared subset of pool CPUs
+  - exclusive allocation of CPU cores
+    * slice off CPU cores from shared subset and assign to workload
+  - mixed allocation of CPU cores
+    * assign both exclusive and shared CPU cores to workload
+  - discover and use kernel-isolated CPU cores
```

Comment:
"discover and use" -> "discovering and using"

Reply:
Updated.